domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -194,13 +193,27 @@ private void statsTracker(String descriptionKey, String sql, long startNanoTime)
   }
 
   /**
-   * Set test instance to use for testing.
-   * If database configuration is already provided, this call is ignored and testing
-   * is performed against the database instance given by configuration.
-   * See {@link org.folio.postgres.testing.PostgresTesterContainer#PostgresTesterContainer()} }
+   * Set PostgresTester instance to use for testing.
+   *
+   * <p>If database configuration is already provided (DB_* env variables or JSON config file),
+   * this call is ignored and testing is performed against the database instance given by configuration.
+   *
+   * <p>Setting the same PostgresTester instance that is already set does nothing. This allows to reuse
+   * a database that some other test has already created, for example to copy a database with sample
+   * data using the <a href="https://stackoverflow.com/a/876565">TEMPLATE method</a>. See also
+   * <a href="https://www.testcontainers.org/test_framework_integration/manual_lifecycle_control/#singleton-containers">
+   * singleton testcontainers</a>.
+   *
+   * <p>Setting a different PostgresTester instance closes the old PostgresTester instance.
+   *
+   * <p>See {@link org.folio.postgres.testing.PostgresTesterContainer#PostgresTesterContainer()}
+   *
    * @param tester instance to use for testing.
    */
   public static void setPostgresTester(PostgresTester tester) {
+    if (tester == postgresTester) {
+      return;
+    }
     stopPostgresTester();
```

Comment:
I don't like this. starting the tester guarantees a new tester-- with empty data. Whether an old one is closed is irrelevant

We are merely saying that closing the old is not necessary as the system does it.

Reply:
This seems rather unlikely to happen in practice. If each test class is going to invoke this method (in order to allow for independent running) then to utilise this would require a shared instance of `PostgresTester` across all test classes. Have I understood that correctly?

Reply:
ok. I see that it is unlike to "cross" boundaries.. But what's the use case, then??

Reply:
The if-clause has been removed.